qiskit/aqua/algorithms/vq_algorithm.py

```diff
@@ -65,6 +65,10 @@ def __init__(self,
         """
         super().__init__(quantum_instance)
 
+        if optimizer is None:
+            logger.info('No optimizer provided, setting it to SLSPQ.')
+            optimizer = SLSQP()
+
```

Comment:
Since optimizer is not Optional it should not be expected to be None and hence this logic seems a bit out of place